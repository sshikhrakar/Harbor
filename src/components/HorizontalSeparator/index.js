import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

function HorizontalSeparator(props) {
  const {
    color,
    thickness,
  } = props;

  return (
    <View style={{ borderWidth: thickness, borderColor: color, width: '100%' }}>
    </View>
  );
}

HorizontalSeparator.propTypes = {
  color: PropTypes.string,
  thickness: PropTypes.number,
};

HorizontalSeparator.defaultProps = {
  thickness: 1,
  color: 'black',
};

export default HorizontalSeparator;
