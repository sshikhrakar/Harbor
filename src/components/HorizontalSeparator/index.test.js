import React from 'react';
import renderer from 'react-test-renderer';

import HorizontalSeparator from './index';

describe('COMPONENTS: HorizontalSeparator', () => {

  it('should render correctly', () => {
    const tree = renderer.create(
      <HorizontalSeparator
        color='red'
        thickness={ 1 }
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
