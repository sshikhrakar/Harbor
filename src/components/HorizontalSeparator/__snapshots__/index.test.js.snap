// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS: HorizontalSeparator should render correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "red",
      "borderWidth": 1,
      "width": "100%",
    }
  }
/>
`;
